### ISSUE TITLE
Cutout data but also WCS

### ISSUE THREAD

**Issue by HealthyPear** [astropy/regions#580]
If I am not wrong this package is missing the equivalent of 

https://docs.astropy.org/en/stable/nddata/utils.html#saving-a-2d-cutout-to-a-fits-file-with-an-updated-wcs

Am I right?

The `regions.RegionMask.cutout` methods makes a cutout only of the data, but not of the WCS (that might come with it?).

Is this a bug or design choice?

Does it make sense to update that method to use `astropy.nddata.Cutout2D` with e.g. an optional keyword argument?

**Comment by keflavich** [comment]
Agree, this is a missing feature and something I do very often.  I'm surprised there isn't already an open issue about it; I'd swear I'd made one.

Yes, I think the cutout2d approach is a good one.

**Comment by keflavich** [comment]
fwiw, my pattern for this is generally:
```python
    preg = reg.to_pixel(wcs)
    mask = preg.to_mask()
    slc_lg, slc_sm = mask.get_overlap_slices(data.shape)
    ww_sl = wcs[slc_lg]
```
